### PR TITLE
Update `tsconfig.json`

### LINKED ISSUE #6: Typechecking fails with "moduleResolution": "node16" or "nodenext"
### Initial checklist

- [X] I read the [support docs](https://github.com/syntax-tree/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/syntax-tree/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/syntax-tree/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Asyntax-tree&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

mdast-util-directive@2.2.2

### Link to runnable example

https://github.com/jameshfisher/bug-report-remark-node16

### Steps to reproduce

Set `"moduleResolution"` to `"node16"` or `"nodenext"` in `tsconfig.json`.

Run `tsc`.

### Expected behavior

No typechecking errors.

### Actual behavior

```
$ npm run build

> test-remark-node16@1.0.0 build
> tsc

example.ts:10:10 - error TS2367: This comparison appears to be unintentional because the types '"link" | "html" | "strong" | "image" | "text" | "linkReference" | "emphasis" | "delete" | "break" | "footnote" | "inlineCode" | "imageReference" | "footnoteReference"' and '"textDirective"' have no overlap.

10   return p.type === "textDirective";
            ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/mdast-util-directive/index.d.ts:14:69 - error TS2694: Namespace '"/Users/jim/dev/tmp/test-remark-node16/node_modules/mdast-util-to-markdown/lib/types"' has no exported member 'Context'.

14 export type Context = import('mdast-util-to-markdown/lib/types.js').Context
                                                                       ~~~~~~~

node_modules/mdast-util-directive/index.d.ts:17:41 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

17 export type ContainerDirective = import('./complex-types').ContainerDirective
                                           ~~~~~~~~~~~~~~~~~

node_modules/mdast-util-directive/index.d.ts:18:36 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

18 export type LeafDirective = import('./complex-types').LeafDirective
                                      ~~~~~~~~~~~~~~~~~

node_modules/mdast-util-directive/index.d.ts:19:36 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

19 export type TextDirective = import('./complex-types').TextDirective
                                      ~~~~~~~~~~~~~~~~~


Found 5 errors in 2 files.

Errors  Files
     1  example.ts:10
     4  node_modules/mdast-util-directive/index.d.ts:14
```

### Affected runtime and version

Node 16

### Affected package manager and version

npm 8

### Affected OS and version

macOS

### Build and bundle tools

_No response_

### PR DESCRIPTION
**Pull request by jameshfisher**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Fix issue described in https://github.com/syntax-tree/mdast-util-directive/issues/6

### CODE REVIEW

**Review by ChristianMurphy — CHANGES_REQUESTED** [review]
Thanks @jameshfisher!
`node16` module resolution is something making its way through the ecosystem, but hasn't reached here yet.
Thanks for looking into it.

A couple items to get it across the finish line.
1. Could you update the ts config of this project to use node16 resolution (using the same config options elsewhere https://github.com/syntax-tree/mdast-util-assert/blob/8958dff406cb658abc4189568ed7af2c56d9320c/tsconfig.json)
2. Could you look into the type error currently breaking CI? (which may require going to some of this project's dependencies and updating them to support `node16` mode as well)

**Comment by wooorm** [comment]
Yup, you should be able to port this commit: https://github.com/syntax-tree/mdast-util-assert/commit/8958dff406cb658abc4189568ed7af2c56d9320c. And things will work :)

**Comment by jameshfisher** [comment]
@wooorm I've ported that commit, updated some dependencies, and tried to resolve some broken references in imports. It still has lots of type-checking errors though, and I don't know how to resolve them.

(It doesn't help that my laptop explodes and takes 15+ seconds for types to update when making any single-character changes in VS Code. This seems to be associated with instances of `visit` causing some perverse worst-case in type-checking. Any advice on this weird problem appreciated!)

File: index.js [hunk 1]
```diff
@@ -7,11 +7,11 @@
  * @typedef {import('mdast-util-from-markdown').CompileContext} CompileContext
  * @typedef {import('mdast-util-from-markdown').Token} Token
  * @typedef {import('mdast-util-to-markdown/lib/types.js').Handle} ToMarkdownHandle
- * @typedef {import('mdast-util-to-markdown/lib/types.js').Context} Context
+ * @typedef {import('mdast-util-to-markdown/index.js').Context} Context
```
Comment (remcohaszing) [inline index.js]:
```suggestion
 * @typedef {import('mdast-util-to-markdown').Context} Context
```

File: index.js [hunk 2]
```diff
@@ -398,7 +398,7 @@ function fence(node) {
 
   return ':'.repeat(size)
 
-  /** @type {import('unist-util-visit-parents/complex-types').BuildVisitor<Root, Directive>} */
+  /** @type {import('unist-util-visit-parents/complex-types.js').BuildVisitor<Root, Directive>} */
```
Comment (remcohaszing) [inline index.js]:
```suggestion
  /** @type {import('unist-util-visit-parents').BuildVisitor<Root, Directive>} */
```

**Review by remcohaszing — CHANGES_REQUESTED** [review]
Thank you! Just 2 small suggestions

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks!

**Comment by wooorm** [comment]
Released in 2.2.3!

**Pull request merged**